tiatoolbox/models/dataset/classification.py

```diff
@@ -65,254 +73,266 @@ def predefined_preproc_func(dataset_name):
     }
     if dataset_name not in preproc_dict:
         raise ValueError(
-            "Predefined preprocessing for" "dataset `%s` does not exist." % dataset_name
+            f"Predefined preprocessing for dataset `{dataset_name}` does not exist."
         )
 
     preproc_list = preproc_dict[dataset_name]
     preproc_func = _TorchPreprocCaller(preproc_list)
     return preproc_func
 
 
-class __ABC_Dataset(torch.utils.data.Dataset):
-    """Defines abstract base class for patch dataset.
+class PatchDataset(abc.ABCPatchDataset):
+    """Defines a simple patch dataset, which inherits
+    from the torch.utils.data.Dataset class.
 
     Attributes:
-        return_labels (bool, False): `__getitem__` will return both the img and
-        its label.
-                If `label_list` is `None`, `None` is returned
+        input_list: Either a list of patches, where each patch is a ndarray or a list of
+        valid path with its extension be (".jpg", ".jpeg", ".tif", ".tiff", ".png")
+        pointing to an image.
+
+        label_list: List of label for sample at the same index in `input_list` .
+        Default is `None`.
```

Comment:
Suggest removing _list from the arument names and just making them plural instead. The fact that it should be a list should be clarified by type annotations and the docstring.

'input_list' -> 'inputs' or 'patches'
'label_list' -> 'labels'

Reply:
Also could possibly use x and y similarly to sklearn / model fitting math convention.